diff --git a/src/borg/archive.py b/src/borg/archive.py
index 7e1272d9..806c6555 100644
--- a/src/borg/archive.py
+++ b/src/borg/archive.py
@@ -141,7 +141,7 @@ def usize_fmt(self):
     def csize_fmt(self):
         return format_file_size(self.csize, iec=self.iec)
 
-    def show_progress(self, item=None, final=False, stream=None, dt=None):
+    def show_progress(self, item=None, final=False, stream=sys.stderr, dt=None):
         now = time.monotonic()
         if dt is None or now - self.last_progress > dt:
             self.last_progress = now
@@ -157,7 +157,14 @@ def show_progress(self, item=None, final=False, stream=None, dt=None):
                     'finished': final,
                 })
                 msg = json.dumps(data)
-                end = '\n'
+                end = "\n"
+            elif not stream.isatty():
+                if not final:
+                    msg = "{0.osize_fmt} O {0.csize_fmt} C {0.usize_fmt} D {0.nfiles} N ".format(self)
+                    msg += remove_surrogates(item.path) if item else ""
+                else:
+                    msg = ""
+                end = "\n"
             else:
                 columns, lines = get_terminal_size()
                 if not final:
@@ -170,9 +177,9 @@ def show_progress(self, item=None, final=False, stream=None, dt=None):
                     if space >= 8:
                         msg += ellipsis_truncate(path, space)
                 else:
-                    msg = ' ' * columns
-                end = '\r'
-            print(msg, end=end, file=stream or sys.stderr, flush=True)
+                    msg = " " * columns
+                end = "\r"
+            print(msg, end=end, file=stream, flush=True)
 
 
 def is_special(mode):
